pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -486,13 +481,36 @@ func translateCORSPolicy(in *networking.CorsPolicy) *route.CorsPolicy {
 	return &out
 }
 
+// getRouteOperation returns readable route description for trace.
+func getRouteOperation(in *route.Route, vsName string, port int) string {
+	path := "/*"
+	m := in.GetMatch()
+	ps := m.GetPathSpecifier()
+	if ps != nil {
+		switch ps.(type) {
+		case *route.RouteMatch_Prefix:
+			path = fmt.Sprintf("%s*", m.GetPrefix())
+		case *route.RouteMatch_Path:
+			path = m.GetPath()
+		case *route.RouteMatch_Regex:
+			path = m.GetRegex()
+		}
+	}
+
+	operation := fmt.Sprintf("%s:%d%s", vsName, port, path)
+	// use host:port + match uri if there is only one destination cluster.
+	wc := in.GetRoute().GetWeightedClusters()
+	if c := wc.GetClusters(); len(c) == 1 {
+		_, _, h, p := model.ParseSubsetKey(c[0].Name)
+		operation = fmt.Sprintf("%s:%d%s", h, p, path)
```

Comment:
Why do you need this ? This is effectively undoing the idea of using a business logic specific operation name.

Reply:
No, this is just to extract the logic out to a function which derives operation name based on envoy route. It returns business logic specific operation name.